src/block/icon-list/edit.js

```diff
@@ -243,12 +241,12 @@ const Edit = props => {
 
 		// Different icon sizes per device preview.
 		const { __experimentalGetPreviewDeviceType: getPreviewDeviceType } = select( 'core/edit-post' )
-		const previewDeviceType = getPreviewDeviceType()
+		const previewDeviceType = getPreviewDeviceType && getPreviewDeviceType()
 		const currentIconSize = (
-			previewDeviceType === 'Desktop' ? props.attributes.iconSize :
+			previewDeviceType && ( previewDeviceType === 'Desktop' ? props.attributes.iconSize :
 				previewDeviceType === 'Tablet' ? ( props.attributes.iconTabletSize || props.attributes.iconSize ) :
-					( props.attributes.iconMobileSize || props.attributes.iconTabletSize || props.attributes.iconSize )
-		) || 20
+					( props.attributes.iconMobileSize || props.attributes.iconTabletSize || props.attributes.iconSize ) )
+		) || props.attributes.iconSize || 20
```

Comment:
In connection with above, you can remove the check for `previewDeviceType` since it's now assigned as `Desktop` if the function is not available